@xen-orchestra/fs/src/fs.spec.js

```diff
@@ -0,0 +1,127 @@
+/* eslint-env jest */
+
+import { getHandler } from '.'
+import { tmpdir } from 'os'
+
+// https://gist.github.com/julien-f/3228c3f34fdac01ade09
+const unsecureRandomBytes = n => {
+  const bytes = Buffer.alloc(n)
+
+  const odd = n & 1
+  for (let i = 0, m = n - odd; i < m; i += 2) {
+    bytes.writeUInt16BE((Math.random() * 65536) | 0, i)
+  }
+
+  if (odd) {
+    bytes.writeUInt8((Math.random() * 256) | 0, n - 1)
+  }
+
+  return bytes
+}
+
+const TEST_DATA = unsecureRandomBytes(1024)
+
+;[`file://${tmpdir()}`].forEach(url => {
+  describe(url, () => {
+    let handler
+    const testDir = `xo-fs-tests-${Date.now()}`
+    const testFile = `${testDir}/file`
+
+    beforeAll(() => {
+      handler = getHandler({ url })
+    })
+    afterAll(() => {
+      handler = undefined
+    })
+
+    beforeEach(async () => {
+      await handler.outputFile(testFile, TEST_DATA)
+    })
+    afterEach(async () => {
+      await handler.rmdir(testDir, { recursive: true }).catch(error => {
+        if (error.code !== 'ENOENT') {
+          throw error
+        }
+      })
+    })
+
+    describe('test handler', () => {
+      it(`handler doesn't crash`, async () => {
+        expect(await handler.test()).toMatchSnapshot()
+      })
+    })
+
+    describe('outputFile', () => {
+      it('writes data to a file', async () => {
+        expect(await handler.readFile(testFile)).toEqual(TEST_DATA)
+      })
+    })
+
+    describe('list', () => {
+      it(`should list the content of folder`, async () => {
+        await expect(await handler.list(testDir)).toEqual(['file'])
+      })
+    })
+
+    describe('createReadStream', () => {
+      it(`should return a stream`, async () => {
+        const stream = await handler.createReadStream(testFile)
+
+        await expect(stream.path).toEqual(`${tmpdir}/${testFile}`)
+      })
```

Comment:
We need to test that reading the whole file will get us `TEST_DATA`.

You can use `get-stream` for this.